include/proxy-wasm/proxy_action.h

```diff
@@ -0,0 +1,27 @@
+// Copyright 2016-2019 Envoy Project Authors
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+namespace proxy_wasm {
+
+// TODO: use the SDK version when it is updated.
+enum class ProxyAction : uint32_t {
+  Illegal = 0,
```

Comment:
Is this expressive enough to capture the range of buffering behaviors that existing HTTP filters can express when returning from something like `decodeHeaders()` or `decodeBody()`?

Reply:
It's not, there is a comment from me on this elsewhere.

Reply:
We are going to be pulling this from the SDK repo, but that has not been updated yet.  This is a placeholder till the spec review completes and then the SDK is updated.  Added comments.